yb-voyager/cmd/importData.go

```diff
@@ -577,7 +577,54 @@ func importData(importFileTasks []*ImportFileTask) {
 	if importerRole == TARGET_DB_IMPORTER_ROLE {
 		importDataCompletedEvent := createSnapshotImportCompletedEvent()
 		controlPlane.SnapshotImportCompleted(&importDataCompletedEvent)
+		packAndSendImportDataPayload(COMPLETE) // TODO: later for other import data commands
+	}
+
+}
+
+func packAndSendImportDataPayload(status string) {
+	//TODO send this for INPROGRESS status in some fixed interval for long running import data
+	if !callhome.SendDiagnostics {
+		return
+	}
+	payload := createCallhomePayload()
+	switch importType {
+	case SNAPSHOT_ONLY:
+		payload.MigrationType = OFFLINE
+	case SNAPSHOT_AND_CHANGES:
+		payload.MigrationType = LIVE_MIGRATION
+	}
+	bytes, err := json.Marshal(targetDBDetails)
+	if err != nil {
+		log.Errorf("error in parsing sourcedb details: %v", err)
+	}
+	payload.TargetDBDetails = string(bytes)
+	payload.MigrationPhase = IMPORT_DATA_PHASE
+	importDataPayload := callhome.ImportDataPhasePayload{
+		ParallelJobs: int64(tconf.Parallelism),
+		StartClean:   bool(startClean),
 	}
+	importRowsMap, err := getImportedSnapshotRowsMap("target")
```

Comment:
We should try and avoid hitting DB from these functions. (because of what Hemant had said about exit handlers potentially getting stuck or taking too long). That could happen in this case if it takes too long to get data from the DB. 

Would it be easy to capture the relevant information and store it in our StatsReporter, and this function can just fetch it in memory? 

(I could only find this case where we are hitting a DB during the packAndSend* functions. If there are other cases, let's discuss those as well)

Reply:
`state.getImportedSnapshotRowsMap` is not hitting DB it's only using local importDataState `.D/.P...` files only.

> (I could only find this case where we are hitting a DB during the packAndSend* functions. If there are other cases, let's discuss those as well)

No there are no other cases where I am calling in these functions, all the DB-related calls are outside of this function in normal flow and then using that info here only if that is populated. Even in this case, this is just local files if there won't be any files or any negative case it will just return 0.